jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassURLHelperImpl.java

```diff
@@ -47,12 +52,28 @@
 		initializeShareableClassloader(loader);
 	}
 	
-	private static native void init(); 
+	private static native void init();
+	
+	/* The passed-in URL path contains a jar file */
+	private boolean newJarFileCheck(URL convertedjarPath) {
+		boolean newJarFile = false;
+		String protocol = convertedjarPath.getProtocol();
+		String jarFile = convertedjarPath.getFile();
+		if ((protocol.equalsIgnoreCase("file") || protocol.equalsIgnoreCase("jar"))  //$NON-NLS-1$ //$NON-NLS-2$
+		&& jarFile.contains(".jar") //$NON-NLS-1$
```

Comment:
Looking at the native code, for the "file" protocol, the name needs to end with ".jar" or ".zip", or contain `!/` or `!\`. 

https://github.com/eclipse/openj9/blob/7a47f42ce12bbb185ff69c7f625fda53addac6ca/runtime/jcl/common/shared.c#L564

Reply:
Already modified the code against the native function `getCpeTypeForProtocol`